pkg/util/array.go

```diff
@@ -30,6 +31,17 @@ func InArray(needle string, haystack []string) bool {
 	return false
 }
 
+// InArrayWithRegexp checks whether the needle can be matched by haystack
+func InArrayWithRegexp(needle string, haystack []string) bool {
+	for _, item := range haystack {
+		matched, _ := regexp.MatchString(item, needle)
+		if matched {
```

Comment:
I think
`if item == needle || matched {`
will better

thanks a lot for your efforts